base/collection_n1ql_common.go

```diff
@@ -335,7 +335,7 @@ func getIndexMetaWithoutRetry(store N1QLStore, indexName string) (exists bool, m
 
 // DropIndex drops the specified index from the current bucket.
 func DropIndex(store N1QLStore, indexName string) error {
-	statement := fmt.Sprintf("DROP INDEX %s.`%s`", store.EscapedKeyspace(), indexName)
+	statement := fmt.Sprintf("DROP INDEX `%s` ON %s", indexName, store.EscapedKeyspace())
```

Comment:
The `ON` syntax is only available in Couchbase Server 7.0+ - so we might need special handling only when using collections for this. Or find a way to make the old syntax work with bucket and collection keyspaces.

https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/dropindex.html#keyspace-ref

Reply:
I changed this to the docs and spec for 6.0 - the reason this didn't get hit by tests earlier is that typically there were no  indexes in the way that the code was created, so I added a test at the end to Drop them after we created them explicitly in the test.